CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing Guidelines
+
+Thanks for thinking about contributing to Project Calico! The success of an
+open source project is entirely down to the efforts of its contributors, so we
+do genuinely want to thank you for contributing.
+
+This document contains some guidance and steps for contributing. Make sure you
+read it thoroughly, because if you miss some of these steps we will have to ask
```

Comment:
This line is also specific to the contribution portion, but this doc talks about other things too like contacting us and raising github issues, not sure if this sentence fits here.